### PR TITLE
[SPARK-14142][SQL] Replace internal use of unionAll with union

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

unionAll has been deprecated in SPARK-14088.
## How was this patch tested?

Should be covered by all existing tests.

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -1361,8 +1361,8 @@ class Dataset[T] private[sql](
    * by a [[distinct]].
    *
    * @group typedrel
-   * @since 2.0.0
-   */
+  * @since 2.0.0
```
Comment (davies) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: ?

**Comment by davies** [comment]
LGTM, one minor comment

**Comment by rxin** [comment]
Merging in master.

**Pull request closed without merging**